### PR TITLE
[INLONG-6322][Sort] Fix write data incorrect for doris connector with sink multiple scenario

### LINKED ISSUE #6322: [Bug][Sort] Fix write data incorrect for doris connector with  sink multiple scenario
### What happened

Write data incorrect for doris connector with  sink multiple scenario：
1.Some tables can be written correctly
2.Some tables cannot be written correctly
3.Some table write duplicates
After troubleshoot, it was found that the reason was: remove the key while traversing the cache map: 'batchMap'.

### What you expected to happen

Write data all right for doris connector with  sink multiple scenario.

### How to reproduce

Write data of multiple database and tables with the doris connector can be reproduced.

### Environment

_No response_

### InLong version

master

### InLong Component

InLong Sort

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by yunqingmoswu**
### Prepare a Pull Request
*(Change the title refer to the following example)*

Title: [INLONG-6322][Sort] Fix write data incorrect for doris connector with sink multiple scenario

*(The following *XYZ* should be replaced by the actual [GitHub Issue](https://github.com/apache/inlong/issues) number)*

Fixes #6322 

### Motivation

Fix write data incorrect for doris connector with sink multiple scenario

### Modifications

Update DorisDynamicSchemaOutputFormat to fix it.

### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [x] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a follow-up issue for adding the documentation

### CODE REVIEW

**Review by healchow — APPROVED** [review]

File: inlong-sort/sort-connectors/doris/src/main/java/org/apache/inlong/sort/doris/table/DorisDynamicSchemaOutputFormat.java [hunk 1]
```diff
@@ -258,35 +265,52 @@ public synchronized void close() throws IOException {
 
     @SuppressWarnings({"rawtypes"})
     public synchronized void flush() {
+        flushing = true;
         if (batchMap.isEmpty()) {
```
Comment (kuansix) [inline inlong-sort/sort-connectors/doris/src/main/java/org/apache/inlong/sort/doris/table/DorisDynamicSchemaOutputFormat.java:269]: need flushing = true; before return

File: inlong-sort/sort-connectors/doris/src/main/java/org/apache/inlong/sort/doris/table/DorisDynamicSchemaOutputFormat.java [hunk 2]
```diff
@@ -258,35 +265,52 @@ public synchronized void close() throws IOException {
 
     @SuppressWarnings({"rawtypes"})
     public synchronized void flush() {
+        flushing = true;
```
Comment (kuansix) [inline inlong-sort/sort-connectors/doris/src/main/java/org/apache/inlong/sort/doris/table/DorisDynamicSchemaOutputFormat.java:268]: volatile can't ensure CAS atomic operation?

Comment (yunqingmoswu) [reply inlong-sort/sort-connectors/doris/src/main/java/org/apache/inlong/sort/doris/table/DorisDynamicSchemaOutputFormat.java:268]: The entire flush needs to ensure atomicity here.

File: inlong-sort/sort-connectors/doris/src/main/java/org/apache/inlong/sort/doris/table/DorisDynamicSchemaOutputFormat.java (hunk 1 above)
Comment (yunqingmoswu) [reply inlong-sort/sort-connectors/doris/src/main/java/org/apache/inlong/sort/doris/table/DorisDynamicSchemaOutputFormat.java:269]: done

**Review by gong — APPROVED** [review]

**Review by thesumery — APPROVED** [review]

File: inlong-sort/sort-connectors/doris/src/main/java/org/apache/inlong/sort/doris/table/DorisDynamicSchemaOutputFormat.java [hunk 3]
```diff
@@ -258,35 +263,68 @@ public synchronized void close() throws IOException {
...
+                                    kvs.getKey(), loadValue), e);
                 }
-                batchMap.remove(kvs.getKey());
+                errorTables.add(kvs.getKey());
+                LOG.warn("The tableIdentifier: {} load failed and the data will be throw away in the future"
+                        + " because the option 'sink.multiple.ignore-single-table-errors' is 'true'", kvs.getKey());
             }
         }
+        if (!errorTables.isEmpty()) {
+            // Clean the key that has erros
```
Comment (EMsnap) [inline inlong-sort/sort-connectors/doris/src/main/java/org/apache/inlong/sort/doris/table/DorisDynamicSchemaOutputFormat.java:298]: errors

**Review by EMsnap — APPROVED** [review]

**Pull request merged**